hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java

```diff
@@ -375,10 +378,12 @@ protected PurgePathRequest prepareDeleteDirRequest(
     }
 
     // step-2: get all sub files under the deletedDir
-    List<OmKeyInfo> subFiles = keyManager
-        .getPendingDeletionSubFiles(volumeId, bucketId,
-            pendingDeletedDirInfo, remainNum);
+    DeleteKeysResult subFileDeleteResult =
+        keyManager.getPendingDeletionSubFiles(volumeId, bucketId,
+            pendingDeletedDirInfo, remainingBufLimit);
+    List<OmKeyInfo> subFiles = subFileDeleteResult.getKeysToDelete();
     remainNum = remainNum - subFiles.size();
+    remainingBufLimit -= subDirDeleteResult.getConsumedSize();
```

Comment:
we can remove remainNum is not used ....

Reply:
done.